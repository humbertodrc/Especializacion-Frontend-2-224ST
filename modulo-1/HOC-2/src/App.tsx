import './App.css'
import { Navbar } from './components/Navbar'
// import { ProductList } from './components/ProductList'
import {useContext} from 'react'
import { ThemeContext } from './context/ThemeContext'
import ProductListHOC from "./components/ProductList"

function App() {

  const {theme} = useContext(ThemeContext)

  return (
    <>
      <Navbar />
      <main className={`p-12 h-screen ${theme === 'light' ? 'bg-white' : 'bg-slate-900'}`}>
        <ProductListHOC data={[]} />
      </main>
    </>
  )
}

export default App
