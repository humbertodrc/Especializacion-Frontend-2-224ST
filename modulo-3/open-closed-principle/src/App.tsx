import {Title} from "./components/Title";
import WithButton from "./components/WithButton/WithButton";
import { WithLink } from "./components/WithLink";

function App() {

  const onCLick = () => {
    console.log("withButton")
  }

	return (
		<div className="App">
			<Title title='Hola Camada 1 🙂❤'>
				<WithButton onClick={onCLick} text="Click me" />
			</Title>

			<Title title='Hola Camada 2 🙂❤'>
				<WithLink href="https://www.digitalhouse.com" text="Digital House" />
			</Title>
		</div>
	);
}

export default App;
