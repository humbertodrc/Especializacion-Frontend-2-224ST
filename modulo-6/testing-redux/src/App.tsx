import React, {useState} from "react";
import {useDispatch, useSelector} from "react-redux";
import {loginUser} from "./store/authSlice";
import {AppDispatch, RootState} from "./store/store"; // Importa los tipos AppDispatch y RootState

function App() {
	// Definición de estados locales para email y password
	const [email, setEmail] = useState<string>("");
	const [password, setPassword] = useState<string>("");

	// Especifica el tipo de dispatch utilizando AppDispatch
	const dispatch = useDispatch<AppDispatch>();

	// Utiliza un selector para obtener el estado de autenticación desde el estado global
	const auth = useSelector((state: RootState) => state.auth);

	// Manejador del submit del formulario de login
	const handleSubmit = (e: React.FormEvent) => {
		// Previene el comportamiento por defecto del formulario
		e.preventDefault();

		// Despacha la acción loginUser con las credenciales del usuario
		dispatch(loginUser({email, password}));
	};

	return (
		<div className='min-h-screen flex items-center justify-center bg-gray-100'>
			<div className='bg-white p-8 rounded-lg shadow-md w-96'>
				<h2 className='text-2xl font-semibold text-center mb-6'>Login</h2>
				{auth.status === "loading" && <p>Processing...</p>}
				{auth.status === "failed" && (
					<p className='text-red-500 text-center mt-4'>Login failed</p>
				)}
				{auth.status === "succeeded" && <p>Welcome</p>}
				<form onSubmit={handleSubmit}>
					<div className='mb-4'>
						<label
							className='block text-gray-700 text-sm font-bold mb-2'
							htmlFor='email'
						>
							Email
						</label>
						<input
							id='email'
							type='email'
							placeholder='Email'
							value={email}
							onChange={(e) => setEmail(e.target.value)}
							className='shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline'
						/>
					</div>
					<div className='mb-6'>
						<label
							className='block text-gray-700 text-sm font-bold mb-2'
							htmlFor='password'
						>
							Password
						</label>
						<input
							id='password'
							type='password'
							placeholder='Password'
							value={password}
							onChange={(e) => setPassword(e.target.value)}
							className='shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline'
						/>
					</div>
					<div className='flex items-center justify-between'>
						<button
							type='submit'
							className={`bg-indigo-500 hover:bg-indigo-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline flex items-center ${
								auth.status === "loading" && "cursor-not-allowed opacity-50"
							}`}
							disabled={auth.status === "loading"}
						>
							{auth.status === "loading" ? (
								<svg
									className='animate-spin h-5 w-5 mr-3 text-white'
									xmlns='http://www.w3.org/2000/svg'
									fill='none'
									viewBox='0 0 24 24'
								>
									<circle
										className='opacity-25'
										cx='12'
										cy='12'
										r='10'
										stroke='currentColor'
										strokeWidth='4'
									></circle>
									<path
										className='opacity-75'
										fill='currentColor'
										d='M4 12a8 8 0 018-8v8z'
									></path>
								</svg>
							) : (
								"Login"
							)}
						</button>
					</div>
				</form>
			</div>
		</div>
	);
}

export default App;

