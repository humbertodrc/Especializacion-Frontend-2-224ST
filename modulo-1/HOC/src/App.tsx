import './App.css'
import { Section } from './components/Section'
import { WithColorSwap } from "./hoc/WithColorSwap"

const SectionWithHOC = WithColorSwap(Section)

function App() {

  return (
    <div className="App">
      <SectionWithHOC title='Higher Order Component' description='Conociendo los HOC' />
      <Section title='Higher Order Component' description='Conociendo los HOC' />
    </div>
  )
}

export default App
