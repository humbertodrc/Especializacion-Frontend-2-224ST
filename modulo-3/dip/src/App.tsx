import "./App.css";
import ConnetedLoginForm from "./components/ConnetedLoginForm";
import {LoginForm} from "./components/LoginForm";

function App() {
	return (
		<div className="App">
			{/* <LoginForm /> */}
			<ConnetedLoginForm />
		</div>
	);
}

export default App;
