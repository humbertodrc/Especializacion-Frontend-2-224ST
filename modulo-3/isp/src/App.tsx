import "./App.css";
import {VideoList} from "./components/VideoList";
import {liveStream, videoThumbnail} from "./data/data";

function App() {
	return (
		<div className="App">
			<h1>Principio de segregaci√≥n de interfaz (ISP)</h1>
			<VideoList items={videoThumbnail} />
			<VideoList items={liveStream} />
		</div>
	);
}

export default App;
