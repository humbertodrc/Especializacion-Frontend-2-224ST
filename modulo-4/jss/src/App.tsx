import "./App.css";

import logo from "./assets/logo.svg";
import { Button } from "./components/Button";
import {Character} from "./components/Character";
import {Modal} from "./components/Modal";
import useModal from "./hooks/useModal";

function App() {
	const {isShowing: isAliveShowed, toggle: toggleAlive} = useModal();
	const {isShowing: isDeadShowed, toggle: toggleDead} = useModal();

	const isActive = true;

	return (
		<div className="App">
			<img src={logo} />
			<div>
				{/* Cambia estilos a JSS */}
				{/* <button onClick={toggleAlive}>
					<p className="ButtonAliveRicks">Rick's vivos</p>
				</button> */}
				<Button
					backGroundColor="#00beab"
					spacing="10px 20px"
					fontSize="16px"
					fontStyle="italic"
					labelColor="#fff"
					fontWeight="bold"
					onClick={toggleAlive}
				>
					<p className="ButtonAliveRicks">Rick's vivos</p>
				</Button>

				{/* Cambia estilos a JSS */}
				{/* <button onClick={toggleDead}>
					<p className="ButtonDeadsRicks">Rick's vivos</p>
				</button> */}
				<Button
					backGroundColor="#ff0000"
					spacing="10px 20px"
					fontSize="16px"
					fontStyle="italic"
					labelColor="#fff"
					fontWeight="bold"
					onClick={toggleDead}
				>
					<p className="ButtonDeadsRicks">Rick's muertos</p>
				</Button>
			</div>

			<Modal isShowing={isAliveShowed} hide={toggleAlive} title="Rick's vivos">
				{/* Cambia estilos a JSS */}
				<Character
					status="alive"
					// borderRadius="30px"
					width="400px"
					height="auto"
					padding="16px"
					textAlign="center"
				/>
			</Modal>

			<Modal isShowing={isDeadShowed} hide={toggleDead} title="Rick's muertos">
				{/* Cambia estilos a JSS */}
				<Character
					status="dead"
					// borderRadius="16px"
					width="400px"
					height="auto"
					padding="16px"
					textAlign="center"
				/>
			</Modal>
		</div>
	);
}

export default App;
