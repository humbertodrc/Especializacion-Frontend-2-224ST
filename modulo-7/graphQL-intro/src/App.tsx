import {useQuery} from "@apollo/client";
import "./App.css";
import {GET_CHARACTERS, GET_CHARACTER_BY_ID} from "./querys";
import {CharacterData, CharactersData} from "./types/character.types";
import {useState} from "react";

function App() {
	const [id, setId] = useState<number | null>(null);
	const {loading, data, error} = useQuery<CharactersData>(GET_CHARACTERS);
	const {
		loading: isLoading,
		data: character,
		error: hasError,
	} = useQuery<CharacterData>(GET_CHARACTER_BY_ID, {
		variables: {
			id,
		},
		// Omitimos la ejecución de la consulta si id no está definido
		skip: !id,
	});

	const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		const value = parseInt(e.target.value);
		setId(isNaN(value) ? null : value);
	};

	console.log(character);

	return (
		<>
			<h1>GraphQL Intro</h1>
			<div>
				<label>
					Character ID:
					<input value={id || ""} onChange={handleChange} />
				</label>
				{isLoading && <p>Loading...</p>}
				{hasError && <p>Error</p>}
				<div className="container">
					{character && character.character && (
						<div className="card" key={character.character.id}>
							<img
								src={character.character.image}
								alt={character.character.name}
							/>
							<div>
								<h2>{character.character.name}</h2>
								<p>
									{character.character.status} - {character.character.species}
								</p>
							</div>
						</div>
					)}
				</div>
			</div>
			<div className="container">
				{loading && <p>Loading...</p>}
				{error && <p>Error</p>}
				{data &&
					data.characters &&
					data.characters.results.map((character) => (
						<div className="card" key={character.id}>
							<img src={character.image} alt={character.name} />
							<div>
								<h2>{character.name}</h2>
								<p>
									{character.status} - {character.species}
								</p>
							</div>
						</div>
					))}
			</div>
		</>
	);
}

export default App;
