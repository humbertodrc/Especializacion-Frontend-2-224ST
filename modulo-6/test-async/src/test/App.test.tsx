import {render, screen, fireEvent} from "@testing-library/react";
import App from "../App";

describe("Pruebas en <App />", () => {
	test("Debe renderizar el componente", () => {
		// Arrange
		render(<App />);

		// Act
		const title = screen.getByText(/Testing Async/i);
		const input = screen.getByPlaceholderText(/Search/i);
		const button = screen.getByText(/Search/i);

		// Assert
		expect(title).toBeInTheDocument();
		expect(input).toBeInTheDocument();
		expect(button).toBeInTheDocument();
	});

	test("Debe mostar los personajes", async () => {
		// Arrange
		render(<App />);

		// Act
		const rick = await screen.findByText(/Rick/i);
		const morty = await screen.findByText(/Morty/i);

		// Assert
		expect(rick).toBeInTheDocument();
		expect(morty).toBeInTheDocument();

		// screen.debug();
	});

	test("Usuario debe poder buscar un personaje", async () => {
		// Arrange
		render(<App />);

		// Act
		const input = screen.getByPlaceholderText(/Search/i);
		fireEvent.change(input, {target: {value: "Rick"}});
		const button = screen.getByText(/Search/i);
		fireEvent.click(button);

		const rick = await screen.findByText(/Rick/i);

		// Assert
		expect(rick).toBeInTheDocument();
	});

	test("Debe hacer match con el snapshot", async () => {
		// Arrange
		const {container} = render(<App />);

		// Act
		const rick = await screen.findByText(/Rick/i);

		// Assert
		expect(rick).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
