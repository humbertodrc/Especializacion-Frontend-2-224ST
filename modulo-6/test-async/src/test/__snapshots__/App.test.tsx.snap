// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      <h1>
        Testing Async
      </h1>
      <div
        class="search"
      >
        <form>
          <input
            placeholder="Search..."
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="grilla"
      >
        <div
          class="card"
        >
          <h2>
            Rick
          </h2>
          <img
            alt="Rick"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
        </div>
        <div
          class="card"
        >
          <h2>
            Morty
          </h2>
          <img
            alt="Morty"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
